rangeserver/src/range_manager.rs

```diff
@@ -564,10 +586,60 @@ where
                 // gets to storage directly. We should implement a memtable to allow us to release
                 // the lock sooner.
                 lock_table.release();
+                // Process transaction complete and remove the requests from the logs
+                self.prefetching_buffer
+                    .process_transaction_complete(tx.id)
+                    .await;
                 Ok(())
             }
         }
     }
+
+    /// Get from database without acquiring any locks
+    /// This is a very basic copy of the 'get' function
+    pub async fn prefetch_get(&self, key: Bytes) -> Result<Option<Bytes>, Error> {
+        let val = self
+            .storage
+            .get(self.range_id, key.clone())
+            .await
+            .map_err(Error::from_storage_error)?;
+        Ok(val)
+    }
+
+    pub async fn prefetch(&self, transaction_id: Uuid, key: Bytes) -> Result<(), Error> {
+        // Request prefetch from the prefetching buffer
+        if let Some(keystate) = self
+            .prefetching_buffer
+            .process_prefetch_request(transaction_id, key.clone())
```

Comment:
I haven't yet looked into the prefetching buffer module itself, but why does this return an option?

Reply:
Originally it was written to be able to return None if there was a failure. I have updated to just return the keystate status directly given the function always returns the key as status requested at a minimum.